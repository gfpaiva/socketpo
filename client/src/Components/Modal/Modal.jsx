import React from 'react';
import PropTypes from 'prop-types';

import './Modal.scss';

const Modal = ({ children, className, ...rest }) => (

	<div
		className={`${className ? `${className} ` : ''}modal`}
		{...rest}
	>
		<div className="modal__wrapper">
			<div className="modal__content">
				{children}
			</div>
		</div>
	</div>
);

Modal.propTypes = {
	className: PropTypes.string,
	children: PropTypes.oneOfType([
		PropTypes.arrayOf(PropTypes.node),
		PropTypes.node
	]).isRequired
};

export default Modal;
