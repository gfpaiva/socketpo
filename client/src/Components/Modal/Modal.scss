.modal {
	background-color: rgba(0, 0, 0, .75);
	height: 100%;
	left: 0;
	position: fixed;
	top: 0;
	width: 100%;

	&__wrapper {
		background-color: #fff;
		left: 50%;
		min-width: 90%;
		padding: 1em;
		position: absolute;
		text-align: center;
		top: 50%;
		transform: translate(-50%, -50%);
	}
}
