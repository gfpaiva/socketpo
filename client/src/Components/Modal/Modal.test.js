import React from 'react';
import { mount } from 'enzyme';
import Modal from './Modal';

describe('<Modal />', () => {
	it('should mount properly with class, props and children', () => {
		const wrapper = mount(
			<Modal
				className="test"
				test="test"
			>
				<p>Test</p>
			</Modal>
		);

		expect(wrapper).toMatchSnapshot();
	});
});
