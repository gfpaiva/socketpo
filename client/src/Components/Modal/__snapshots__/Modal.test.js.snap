// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should mount properly with class, props and children 1`] = `
<Modal
  className="test"
  test="test"
>
  <div
    className="test modal"
    test="test"
  >
    <div
      className="modal__wrapper"
    >
      <div
        className="modal__content"
      >
        <p>
          Test
        </p>
      </div>
    </div>
  </div>
</Modal>
`;
